### PR TITLE
Fix str in dumps

### PR DESCRIPTION
**Pull request by alejoe91**
Fixes dumps when file names are not passed

### CODE REVIEW

File: spikeextractors/baseextractor.py [hunk 1]
```diff
@@ -67,9 +67,9 @@ def dump_to_json(self, file_path=None):
         '''
         if self.check_if_dumpable():
             if file_path is None:
-                if 'Recording' in self.__class__:
+                if 'Recording' in str(self.__class__):
```
Comment (samuelgarcia) [inline spikeextractors/baseextractor.py:70]: or self.__class__.__name__

Comment (alejoe91) [reply spikeextractors/baseextractor.py:70]: that doesn't work

Comment (alejoe91) [reply spikeextractors/baseextractor.py:70]: `type(self).__name__` works as well, or `str(self.__class__)`

Comment (alejoe91) [reply spikeextractors/baseextractor.py]: same as thos type(self).__name__

**Comment by yarikoptic** [comment]
Thanks! I will check when get to the laptop. I hope you don't mind me commenting: it feels like there is just missing class method (or even just an attribute) defined at the top base classes like `get_default_filename` to avoid all the if, and just to rely on inheritance.

**Comment by alejoe91** [comment]
fixed here https://github.com/SpikeInterface/spikeextractors/pull/372

**Pull request closed without merging**